### PR TITLE
[Filebeat] [MongoDB] Support MongoDB 4.4 json logs

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

MongoDB 4.4 changed logs to a structured JSON format, this PR adds support for mongodb logs json format, and support the old format too.

## Why is it important?

to support new logs format in filebeat MongoDB module.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes https://github.com/elastic/beats/issues/20501

### CODE REVIEW

**Comment by jsoriano** [comment]
@tetianakravchenko thanks for this contribution!

[Failure in tests](https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-24774/2/testReport/junit/filebeat.tests.system.test_modules/Test/Build_Test___filebeat_build___test_fileset_file_066_mongodb/) seems related:
```
Pipeline processor configured for non-existent pipeline [filebeat-8.0.0-mongodb-log-pipeline-plaintext]
```

I think the problem is that new pipelines need to be added to the manifest in `filebeat/module/mongodb/log/manifest.yml`, could you please add the pipelines there?

**Comment by tetianakravchenko** [comment]
@jsoriano thank you for the feedback! I've added new pipelines to the manifest https://github.com/elastic/beats/pull/24774/commits/9f5d34cfd8d2b97fd828f5eb9a165ce952e61356

File: filebeat/module/mongodb/_meta/docs.asciidoc [hunk 1]
```diff
@@ -13,7 +13,7 @@ include::../include/gs-link.asciidoc[]
 [float]
 === Compatibility
 
-The +{modulename}+ module was tested with logs from versions v3.2.11 on Debian.
+The +{modulename}+ module was tested with plaintext logs from version v3.2.11 on Debian and json logs from version v4.4.4 on Ubuntu.
```
Comment (jsoriano) [inline filebeat/module/mongodb/_meta/docs.asciidoc:16]: Thanks for updating the docs! You will need to run `mage update` from the `filebeat` directory and commit the changes after these changes.

Comment (tetianakravchenko) [reply filebeat/module/mongodb/_meta/docs.asciidoc:16]: done https://github.com/elastic/beats/pull/24774/commits/9823ac8a8ce4976a3cb6ff2b606dff62de67108b

**Comment by cihantunali** [comment]
Hi there,
I dont know if it helps or not but I created a GROK pattern to parse it. I tried to implement it via manuel way to filebeat (created new pipeline and forced filebeat to use it) but no luck :( 

{"t":{"\$date":"%{TIMESTAMP_ISO8601:timestamp}"},"s":"%{WORD:severity}",%{SPACE}"c":"%{WORD:component}",%{SPACE}"id":%{INT:id},%{SPACE}"ctx":"%{DATA:context}","msg":"%{GREEDYDATA:message}

I hope, release come soon. Thanks!

**Comment by cihantunali** [comment]
Hi there!
I hope it can build. Thank you @jsoriano for your effort.

**Comment by tetianakravchenko** [comment]
Hi @jsoriano, I see that build is failing mainly with timeout error:
```
E               beat.beat.TimeoutError: Timeout waiting for 'cond' to be true. Waited 10 seconds.
```
or 
```
E           Failed: Timeout >90.0s
```
is there anything I can do here?

**Comment by jsoriano** [comment]
@tetianakravchenko try updating your branch with master, failing tests can be related to outdated files.

**Comment by jsoriano** [comment]
I have updated the branch, let's see if tests pass now :crossed_fingers:

**Comment by cihantunali** [comment]
I think there is same error here

E           Failed: Timeout >90.0s

**Comment by jsoriano** [comment]
I am not sure why other modules fail, but there seems to be a problem with the change in the MongoDB module. Running the tests locally I see this error in filebeat logs:
```
2021-06-07T07:17:32.045-0200    ERROR   [publisher_pipeline_output]     pipeline/output.go:154  Failed to connect to backoff(elasticsearch(http://elasticsearch:9200)): Connection marked as failed because the onConnect callback failed: 1 error: error loading pipeline for fileset mongodb/log: couldn't load pipeline: couldn't load json. Error: 400 Bad Request: {"error":{"root_cause":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'ACCESS'","       ^---- HERE"],"script":"mongodb.c == 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":21},"suppressed":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'WRITE'","       ^---- HERE"],"script":"mongodb.c == 'WRITE'","lang":"painless","position":{"offset":7,"start":0,"end":20}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c != 'WRITE' && mongodb. ...","       ^---- HERE"],"script":"mongodb.c != 'WRITE' && mongodb.c != 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":32}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.s == 'F' || mongodb.s == ...","       ^---- HERE"],"script":"mongodb.s == 'F' || mongodb.s == 'E'","lang":"painless","position":{"offset":7,"start":0,"end":32}}]}],"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'ACCESS'","       ^---- HERE"],"script":"mongodb.c == 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":21},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"},"suppressed":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'WRITE'","       ^---- HERE"],"script":"mongodb.c == 'WRITE'","lang":"painless","position":{"offset":7,"start":0,"end":20},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c != 'WRITE' && mongodb. ...","       ^---- HERE"],"script":"mongodb.c != 'WRITE' && mongodb.c != 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":32},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.s == 'F' || mongodb.s == ...","       ^---- HERE"],"script":"mongodb.s == 'F' || mongodb.s == 'E'","lang":"painless","position":{"offset":7,"start":0,"end":32},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.s]"}}]},"status":400}. Response body: {"error":{"root_cause":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'ACCESS'","       ^---- HERE"],"script":"mongodb.c == 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":21},"suppressed":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'WRITE'","       ^---- HERE"],"script":"mongodb.c == 'WRITE'","lang":"painless","position":{"offset":7,"start":0,"end":20}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c != 'WRITE' && mongodb. ...","       ^---- HERE"],"script":"mongodb.c != 'WRITE' && mongodb.c != 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":32}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.s == 'F' || mongodb.s == ...","       ^---- HERE"],"script":"mongodb.s == 'F' || mongodb.s == 'E'","lang":"painless","position":{"offset":7,"start":0,"end":32}}]}],"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'ACCESS'","       ^---- HERE"],"script":"mongodb.c == 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":21},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"},"suppressed":[{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c == 'WRITE'","       ^---- HERE"],"script":"mongodb.c == 'WRITE'","lang":"painless","position":{"offset":7,"start":0,"end":20},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.c != 'WRITE' && mongodb. ...","       ^---- HERE"],"script":"mongodb.c != 'WRITE' && mongodb.c != 'ACCESS'","lang":"painless","position":{"offset":7,"start":0,"end":32},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.c]"}},{"type":"script_exception","reason":"compile error","processor_type":"append","script_stack":["mongodb.s == 'F' || mongodb.s == ...","       ^---- HERE"],"script":"mongodb.s == 'F' || mongodb.s == 'E'","lang":"painless","position":{"offset":7,"start":0,"end":32},"caused_by":{"type":"illegal_argument_exception","reason":"cannot resolve symbol [mongodb.s]"}}]},"status":400}
```

There seems to be a problem with the pipeline.

I can reproduce the error running the following command from the `filebeat` directory: `PYTEST_ADDOPTS="-k test_modules" GENERATE=1 TESTING_FILEBEAT_MODULES=mongodb mage pythonIntegTest`. 

After running the tests, logs are available in `build/system-tests/run/test_modules.Test.test_fileset_file_0_mongodb/output.log`.

@tetianakravchenko can you take a look?

**Comment by tetianakravchenko** [comment]
thank you for the hint, I will have a look 👍

**Comment by tetianakravchenko** [comment]
Hi @jsoriano. Locally I have the command below running successfully after the latest changes
```
filebeat directory: PYTEST_ADDOPTS="-k test_modules" GENERATE=1 TESTING_FILEBEAT_MODULES=mongodb mage pythonIntegTest
```
Could you please trigger tests?

**Comment by jsoriano** [comment]
Thanks @tetianakravchenko! I can confirm that tests pass now locally for me, let's wait for CI :crossed_fingers:

**Review by jsoriano — COMMENTED** [review]
@tetianakravchenko sorry, one last thing, could you please add a changelog entry in the `CHANGELOG.next.asciidoc` file? For the rest it LGTM, thanks!

**Review by jsoriano — APPROVED** [review]
LGTM, thanks!

**Comment by cihantunali** [comment]
G8 work! When it can be released and which version?

**Comment by jsoriano** [comment]
@cihantunali this will be likely available in Filebeat 7.14.0.

**Comment by tetianakravchenko** [comment]
@jsoriano my idea was to keep the original log message, as it contains [`attr` field](https://docs.mongodb.com/manual/reference/log-messages/#json-log-output-format ), where might be stored important information, like `locks`, `readConcern`, `writeConcern`, `command`, etc (all those logs are part of slow logs). As content of this field depends on log type, it doesn't make sense to parse each subfield, but keeping original message with `attr` (other optional fields are `truncated`, `size`, `tags`) for debugging purpose I think would be helpful.

**Comment by jsoriano** [comment]
@tetianakravchenko good point, ok, I will keep the original message but under the ECS `log.original` field. I have updated the PR, would you mind to review it?
Maybe in the future we can add something to store these attributes.

**Comment by jsoriano** [comment]
Change applied to integrations in https://github.com/elastic/integrations/pull/1138.

**Pull request merged**